adapters/repos/db/index.go

```diff
@@ -2483,6 +2484,11 @@ func (i *Index) drop() error {
 
 	i.closingCancel()
 
+	// Check if a backup is in progress. Dont delete files in this case so the backup process can complete successfully
+	// The files will be deleted after the backup is completed and in case of a crash on next startup.
+	lastBackup := i.lastBackup.Load()
+	keepFiles := lastBackup != nil
```

Comment:
is it possible for `i.lastBackup` value to change while `drop` is already in progress?

Reply:
The closelock is help during index drop, I added one missing .Rlock to the backup release and comments at the location that already had the lock. Now there should be no possibility for the value to change during dropping